ros/angel_system_nodes/angel_system_nodes/object_descriptor_generator.py

```diff
@@ -0,0 +1,215 @@
+            pretrained, None, 5, 3
+        )
+
+        backbone = resnet_fpn_backbone(
+            'resnet50',
+            False,
+            trainable_layers=trainable_backbone_layers
+        )
+
+        super().__init__(backbone=backbone, num_classes=91)
+
+        state_dict = load_state_dict_from_url(model_urls['fasterrcnn_resnet50_fpn_coco'],
+                                              progress=True)
+        self.load_state_dict(state_dict)
+        overwrite_eps(self, 0.0)
+
+    def forward(self, images, targets=None):
+        """
+        Overrides the FasterRCNN forward call to return the output of the
+        model backbone.
+
+        Modified version of thetorchvision `GeneralizedRCNN` forward call().
+        """
+        original_image_sizes: List[Tuple[int, int]] = []
+        for img in images:
+            val = img.shape[-2:]
+            torch._assert(
+                len(val) == 2,
+                f"expecting the last two dimensions of the Tensor to be H and W instead got {img.shape[-2:]}",
+            )
+            original_image_sizes.append((val[0], val[1]))
+
+        images, targets = self.transform(images, targets)
+
+        features = self.backbone(images.tensors)
+        return features
+
+
+class DescriptorGenerator(Node):
+    """
+    ROS node that subscribes to `Image` messages and publishes descriptors
+    for those images in the form of `ObjectDescriptors` messages.
+    """
+
+    def __init__(self):
+        torch_device: str = "cpu",
+        super().__init__(self.__class__.__name__)
+
+        self._image_topic = self.declare_parameter("image_topic", "PVFrames").get_parameter_value().string_value
+        self._desc_topic = self.declare_parameter("descriptor_topic", "ObjectDescriptors").get_parameter_value().string_value
+        self._torch_device = self.declare_parameter("torch_device", "cpu").get_parameter_value().string_value
+        self._feature_layer = self.declare_parameter("feature_layer", "head").get_parameter_value().string_value
+
+        # Currently supported layers from torch's resnet_fpn_backbone
+        allowable_layers = ['0', '1', '2', '3', 'pool', 'head']
+        if self._feature_layer not in allowable_layers:
+            raise ValueError(
+                f"Layer {self._feature_layer} not currently supported"
+            )
+
+        log = self.get_logger()
+        log.info(f"Image topic: {self._image_topic}")
+        log.info(f"Torch device? {self._torch_device}")
+        log.info(f"Feature layer? {self._feature_layer}")
+
+        self._model = None
+
+        self._subscription = self.create_subscription(
+            Image,
+            self._image_topic,
+            self.listener_callback,
+            1
+        )
+        self._publisher = self.create_publisher(
+            ObjectDescriptors,
+            self._desc_topic,
+            1
+        )
+
+        self.transforms = transforms.Compose([
+            transforms.ToTensor(),
+        ])
```

Comment:
Since this is just one thing it could just be 
```suggestion
        self.transforms = transforms.ToTensor()
```